pkg/mutation/path/tester/tester.go

```diff
@@ -56,31 +63,57 @@ func isPrefix(short, long *parser.Path) bool {
 	return true
 }
 
-// ValidatePathTests returns whether a set of path tests are valid against the provided location
-func ValidatePathTests(location *parser.Path, pathTests []Test) error {
+// validatePathTests returns whether a set of path tests are valid against the provided location
+func validatePathTests(location parser.Path, pathTests []Test) error {
 	for _, pathTest := range pathTests {
 		if !isPrefix(pathTest.SubPath, location) {
-			return errors.New("all subpaths must be a prefix of the `location` value of the mutation")
+			return fmt.Errorf("%w: subpath %q is not a prefix of location %q", ErrPrefix, pathTest.SubPath, location)
 		}
 	}
-	if _, err := New(pathTests); err != nil {
-		return err
-	}
 	return nil
 }
 
 // New creates a new Tester object
-func New(tests []Test) (*Tester, error) {
+func New(location parser.Path, tests []Test) (*Tester, error) {
+	err := validatePathTests(location, tests)
+	if err != nil {
+		return nil, err
+	}
+
+	paths := make(map[int]*parser.Path)
 	idx := &Tester{
 		tests: make(map[int]Condition),
 	}
+
+	// Read in all tests before checking for conflicts.
+	idxLowestMustNot := math.MaxInt64
```

Comment:
`/mutation/path/tester/tester.go:89:19: constant 9223372036854775807 overflows int`

@willbeason looks like this breaks armv7 builds which are not 64-bit. can we use 32-bit max int, or should we drop armv7, or any other options we have?

Reply:
I'll take a look

Reply:
Can we sidestep this problem by using -1 for a sentinel value?